lib/timers.js

```diff
@@ -481,6 +481,7 @@ Timeout.prototype.unref = function() {
     }
 
     var handle = reuse(this);
+    delete handle._list;
```

Comment:
> Why not removing the reference to the `TimersList` instance in `reuse`'s implementation instead?

Messes with some tests that check thing after `unenroll()`, which also uses `reuse()`.

--

Setting it to `undefined` would be faster but I figured it could be confusing that sometimes there would be a `_list` property.

If that is acceptable we can switch to it though.